docs/addons/index.md

```diff
@@ -8,26 +8,27 @@ Within the context of the `ssp-amazon-eks` framework, an add-on is abstracted as
-| [`VeleroAddOn`](./velero.md) | Adds [Velero](https://velero.io/) to the EKS Cluster |
+| Add-on                                                                           | Description                                                                                                                           |
+|----------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------------------------------|
+| [`AppMeshAddOn`](./app-mesh)                                                     | Adds an AppMesh controller and CRDs (pending validation on the latest version of CDK)                                                 |
+| [`ArgoCDAddOn`](./argo-cd)                                                       | Provisions Argo CD into your cluster.                                                                                                 |
+| [`AWS Load Balancer Controller`](./aws-load-balancer-controller)                 | Provisions the AWS Load Balancer Controller into your cluster.                                                                        |
+| [`CalicoAddOn`](./calico)                                                        | Adds the Calico 1.7.1 CNI/Network policy engine                                                                                       |
+| [`ClusterAutoscalerAddOn`](./cluster-autoscaler)                                 | Adds the standard cluster autoscaler ([Karpenter](https://github.com/awslabs/karpenter) is coming)                                    |
+| [`ContainerInsightsAddOn`](./container-insights)                                 | Adds Container Insights support integrating monitoring with CloudWatch                                                                |
+| [`CoreDnsAddOn`](./coredns.md)                                                   | Adds CoreDNS Amazon EKS add-on. CoreDNS is a flexible, extensible DNS server that can serve as the Kubernetes cluster DNS             |
+| [`KubeProxyAddOn`](./kube-proxy.md)                                              | Adds kube-proxy Amazon EKS add-on. Kube-proxy maintains network rules on each Amazon EC2 node                                         |
+| [`MetricsServerAddOn`](./metrics-server.md)                                      | Adds metrics server (pre-req for HPA and other monitoring tools)                                                                      |
+| [`ExternalDnsAddOn`](./external-dns)                                             | Adds [External DNS](https://github.com/kubernetes-sigs/external-dns) support for AWS to the cluster, integrating with Amazon Route 53 |
+| [`Keptn`](https://github.com/keptn-sandbox/keptn-ssp-addons/blob/main/README.md) | [Keptn](https://www.keptn.sh) AddOn                                                                                                   |
```

Comment:
Nice work on the SSP add-on. A few comments before we apply the changes:

1. We would like to achieve consistency with respect to the helm add-ons for a number of reasons (using private helm repos in private clusters, ability to control configuration through gitops). For that reason we have HelmAddOn base class. Here is an [example](https://github.com/shapirov103/ssp-eks-extension/blob/main/lib/index.ts).
2. It would help to have an example of usage on you README page, e.g. a blueprint with control plane add-on and a blueprint with execution plane add-on (or combined).
3. Do you mind adding a pattern to the https://github.com/aws-samples/ssp-eks-patterns  that shows how to use it. We will use it to validate the functionality. 

For the latter you can see an example with [Kubecost add-on pattern.](https://github.com/aws-samples/ssp-eks-patterns/blob/main/lib/kubecost-construct/index.ts)

Reply:
Thank you for your fast review.

* We changed the implementation to use the HelmAddOn base class
* A quick start has been provided on the README page, links to more detailed documentation is provided there
* We added a pattern to the ssp-eks-patterns repository (PR: https://github.com/aws-samples/ssp-eks-patterns/pull/13)

Reply:
@shapirov103 : Is there anything more needed from our side?